recipes/libxml2/all/conanfile.py

```diff
@@ -56,92 +61,126 @@ class Libxml2Conan(ConanFile):
+        return [name for name in self.info.options.keys() if name not in ["shared", "fPIC", "include_utils"]]
 
     @property
     def _settings_build(self):
         return getattr(self, "settings_build", self.settings)
 
-    @property
-    def _is_msvc(self):
-        return str(self.settings.compiler) in ["Visual Studio", "msvc"]
-
     @property
     def _is_mingw_windows(self):
         return self.settings.compiler == "gcc" and self.settings.os == "Windows" and self._settings_build.os == "Windows"
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
+        if Version(self.version) >= "2.10.3":
+            del self.options.docbook
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
-        del self.settings.compiler.libcxx
-        del self.settings.compiler.cppstd
+            try:
+                del self.options.fPIC
+            except Exception:
+                pass
+        try:
+            del self.settings.compiler.libcxx
+        except Exception:
+            pass
+        try:
+            del self.settings.compiler.cppstd
+        except Exception:
+            pass
 
     def requirements(self):
         if self.options.zlib:
-            self.requires("zlib/1.2.12")
+            self.requires("zlib/1.2.13")
         if self.options.lzma:
             self.requires("xz_utils/5.2.5")
         if self.options.iconv:
             self.requires("libiconv/1.17")
         if self.options.icu:
             self.requires("icu/71.1")
 
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
     def build_requirements(self):
-        if not (self._is_msvc or self._is_mingw_windows):
+        if not (is_msvc(self) or self._is_mingw_windows):
             if self.options.zlib or self.options.lzma or self.options.icu:
-                self.build_requires("pkgconf/1.7.4")
-            if self._settings_build.os == "Windows" and not tools.get_env("CONAN_BASH_PATH"):
-                self.build_requires("msys2/cci.latest")
+                self.tool_requires("pkgconf/1.9.3")
+            if self._settings_build.os == "Windows":
+                self.win_bash = True
+                if not self.conf.get("tools.microsoft.bash:path", default=False, check_type=str):
+                    self.tool_requires("msys2/cci.latest")
 
     def source(self):
-        # can't use strip_root here because if fails since 2.9.10 with:
-        # KeyError: "linkname 'libxml2-2.9.1x/test/relaxng/ambig_name-class.xml' not found"
-        tools.get(**self.conan_data["sources"][self.version])
-        rename(self, "libxml2-{}".format(self.version), self._source_subfolder)
-
-    @contextmanager
-    def _msvc_build_environment(self):
-        with tools.chdir(os.path.join(self._source_subfolder, 'win32')):
-            with tools.vcvars(self.settings):
-                with tools.environment_append(VisualStudioBuildEnvironment(self).vars):
```

Comment:
So `VisualStudioBuildEnvironment` is removed (legacy build helper), but what is the replacement? AFAIK none. So who's in charge to honor settings.build_type, tools.build.cflags etc?

Reply:
Until https://github.com/conan-io/conan/issues/12188 is resolved, I advice to not migrate the msvc branch.

Reply:
@paulharris see https://github.com/conan-io/conan-center-index/pull/13941 for a workaround

Reply:
Great thanks @SpaceIm